include/unifex/receiver_concepts.hpp

```diff
@@ -115,15 +124,32 @@ namespace _rec_cpo {
       return unifex::tag_invoke(
           _set_error_fn{}, (Receiver &&) r, (Error&&) error);
     }
+
+    // direct call
     template(typename Receiver, typename Error)
-      (requires (!tag_invocable<_set_error_fn, Receiver, Error>))
+      (requires (!tag_invocable<_set_error_fn, Receiver, Error>) AND
+         std::invocable<decltype(&Receiver::set_error), Receiver, Error>)
```

Comment:
> ```c++
> std::invocable<decltype(&Receiver::set_error), Receiver, Error>
> ```

This is not the right way to check for a `.set_error(Error)` member function. Consider that `Receiver::set_error` could be a template. In that case, `&Receiver::set_error` is ill-formed. Check how such constraints are checked elsewhere; you'll need to test the validity of the expression `rec.set_error(err)` in a SFINAE context (or with a concept) to get a correct answer.